src/App.js

```diff
@@ -1,25 +1,151 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import Boxes from "./Components/Boxes";
+import "./App.css";
+import ColourPicker from "./Components/ColourPicker";
+
+const App = () => {
+  const [boxes, updateBoxes] = useState([
+    { id: 1, highlight: false, colour: "yellow" },
+    { id: 2, highlight: false, colour: "yellow" },
+    { id: 3, highlight: false, colour: "yellow" },
+    { id: 4, highlight: false, colour: "yellow" },
+    { id: 5, highlight: false, colour: "yellow" },
+    { id: 6, highlight: false, colour: "yellow" },
+    { id: 7, highlight: false, colour: "yellow" },
+    { id: 8, highlight: false, colour: "yellow" },
+    { id: 9, highlight: false, colour: "yellow" },
+  ]);
+
+  const [selectedColour, updateSelectedColour] = useState("yellow");
+
+  const handleHighlight = (id, selectedColour) => {
+    let newBoxes = [...boxes];
+    const boxIndex = id - 1;
+    newBoxes[boxIndex].highlight = !newBoxes[boxIndex].highlight;
+    newBoxes[boxIndex].colour = selectedColour;
+    newBoxes = changeSurrounding(newBoxes, boxIndex, selectedColour);
+    updateBoxes(newBoxes);
+  };
+
+  const handleColourChange = (currentColour, selected) => {
+    let newColour = selected;
+    if (currentColour === newColour) return;
+    updateSelectedColour(newColour);
+  };
+
+  const changeSurrounding = (newBoxes, index, selectedColour) => {
```

Comment:
I did this without the `switch/case` function :-), it will allow you to add in more blocks a lot more easily as well in future

detete all of the `changeSurrounding` function and replace with

```
  const changeGrid = ({ numbers, newBoxes, selectedColour }) => {
    numbers.forEach((number) => {
      newBoxes[number].highlight = !newBoxes[number].highlight;
      newBoxes[number].colour = selectedColour;
    });
    return newBoxes;
  };

  const changeSurrounding = ({ newBoxes, index, selectedColour }) => {
    newBoxes = changeGrid({
      numbers: numbers[index],
      newBoxes,
      selectedColour,
    });
    return newBoxes;
  };
```

change the `boxes` state from

```
{ id: 1, highlight: false, colour: "yellow" },
    { id: 2, highlight: false, colour: "yellow" },
    { id: 3, highlight: false, colour: "yellow" },
    { id: 4, highlight: false, colour: "yellow" },
    { id: 5, highlight: false, colour: "yellow" },
    { id: 6, highlight: false, colour: "yellow" },
    { id: 7, highlight: false, colour: "yellow" },
    { id: 8, highlight: false, colour: "yellow" },
    { id: 9, highlight: false, colour: "yellow" },
```
to

```
    { id: 1, highlight: false, colour: "yellow", numbers: [1, 3] },
    { id: 2, highlight: false, colour: "yellow", numbers: [0, 2, 4] },
    { id: 3, highlight: false, colour: "yellow", numbers: [1, 5] },
    { id: 4, highlight: false, colour: "yellow", numbers: [0, 4, 6] },
    { id: 5, highlight: false, colour: "yellow", numbers: [1, 3, 5, 7] },
    { id: 6, highlight: false, colour: "yellow", numbers: [2, 4, 8] },
    { id: 7, highlight: false, colour: "yellow", numbers: [3, 7] },
    { id: 8, highlight: false, colour: "yellow", numbers: [4, 6, 8] },
    { id: 9, highlight: false, colour: "yellow", numbers: [5, 7] },
```

It works the same, to add more squares to the function just add one to this array and make sure the numbers are included  :D